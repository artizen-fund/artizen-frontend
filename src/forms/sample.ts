export const schema = {
  type: 'object',
  properties: {
    name: {
      type: 'string',
      minLength: 3,
      description: 'Please enter your name',
    },
    vegetarian: {
      type: 'boolean',
    },
    vegetarianSure: {
      type: 'boolean',
    },
    carnivore: {
      type: 'boolean',
    },
    birthDate: {
      type: 'string',
      format: 'date',
    },
    nationality: {
      type: 'string',
      enum: ['', 'DE', 'IT', 'JP', 'US', 'RU', 'Other'],
    },
    iceCream: {
      type: 'string',
      enum: ['vanilla', 'chocolate', 'strawberry', 'neapolitan'],
    },
    personalData: {
      type: 'object',
      properties: {
        age: {
          type: 'integer',
          description: 'Please enter your age.',
        },
        height: {
          type: 'number',
        },
        drivingSkill: {
          type: 'number',
          maximum: 10,
          minimum: 1,
          default: 7,
        },
      },
      required: ['age', 'height'],
    },
    occupation: {
      type: 'string',
    },
    postalCode: {
      type: 'string',
      maxLength: 5,
    },
  },
  required: ['occupation', 'nationality'],
}

export const uischema = {
  type: 'VerticalLayout',
  elements: [
    {
      type: 'HorizontalLayout',
      elements: [
        {
          type: 'Control',
          scope: '#/properties/name',
        },
        {
          type: 'Control',
          scope: '#/properties/vegetarian',
        },
        {
          type: 'Control',
          scope: '#/properties/vegetarianSure',
          options: {
            format: 'checkbox',
            size: 'small',
          },
        },
        {
          type: 'Control',
          scope: '#/properties/carnivore',
          options: {
            format: 'switch',
          },
        },
        {
          type: 'Control',
          scope: '#/properties/iceCream',
          options: {
            format: 'segmented',
          },
        },
        {
          type: 'Control',
          scope: '#/properties/personalData/properties/age',
        },
        {
          type: 'Control',
          scope: '#/properties/birthDate',
        },
      ],
    },
    {
      type: 'Label',
      text: 'Additional Information',
    },
    {
      type: 'HorizontalLayout',
      elements: [
        {
          type: 'Control',
          scope: '#/properties/personalData/properties/height',
        },
        {
          type: 'Control',
          scope: '#/properties/nationality',
        },
        {
          type: 'Control',
          scope: '#/properties/occupation',
          suggestion: [
            'Accountant',
            'Engineer',
            'Freelancer',
            'Journalism',
            'Physician',
            'Student',
            'Teacher',
            'Other',
          ],
        },
      ],
    },
  ],
}

export const initialState = {
  name: undefined,
  vegetarian: undefined,
  vegetarianSure: undefined,
  carnivore: undefined,
  birthDate: undefined,
  iceCream: undefined,
  personalData: {
    age: undefined,
  },
  postalCode: undefined,
}
