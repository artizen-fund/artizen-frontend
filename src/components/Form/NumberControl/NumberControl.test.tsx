import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { NumberControl } from './'
import { schema, uischema } from '@forms/sample'

describe('NumberControl', () => {
  const INPUT_PATH = '#/properties/integerExample'

  it('renders NumberControl unchanged', () => {
    const handleChange = () => console.log('derp')
    const { container } = render(
      <NumberControl
        data={5}
        path={uischema.elements.find(e => e.scope === INPUT_PATH)?.scope!}
        {...{ handleChange, schema, uischema }}
        label="Integer test"
      />,
    )
    expect(container).toMatchSnapshot()
  })
})

/* 
Possible future testsâ€¦

- onchange triggers console log
- error state (min, max, required) is respected
*/
