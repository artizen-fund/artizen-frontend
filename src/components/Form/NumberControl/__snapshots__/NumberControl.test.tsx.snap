// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberControl renders NumberControl unchanged 1`] = `
<div>
  <div
    class="sc-95fbcf7b-0 eTbwSC"
  >
    <div
      class="sc-e8f3f789-0 jYEzXi"
    >
      <input
        class="hasData"
        step="any"
        type="number"
        value="5"
      />
      <label
        class="sc-811bf69e-0 ATCA"
      >
        Integer test
      </label>
      <div
        class="sc-917c067-0 cilYzr"
      />
    </div>
  </div>
</div>
`;
